pyro/distributions/empirical.py

```diff
@@ -11,12 +11,38 @@
 @copy_docs_from(TorchDistribution)
 class Empirical(TorchDistribution):
     r"""
-    Empirical distribution associated with the sampled data.
+    Empirical distribution associated with the sampled data. Note that the shape
+    requirement for `log_weights` is that its shape must match the leftmost shape
+    of `samples`. Samples are aggregated along the ``aggregation_dim``, which is
+    the rightmost dim of `log_weights`.
+
+    Example:
+
+    >>> emp_dist = Empirical(torch.randn(2, 3, 10), torch.ones(2, 3))
+    >>> emp_dist.batch_shape
+    torch.Size([2])
+    >>> emp_dist.event_shape
+    torch.Size([10])
+
+    >>> single_sample = emp_dist.sample()
+    >>> single_sample.shape
+    torch.Size([2, 10])
+    >>> batch_sample = emp_dist.sample((100,))
+    >>> batch_sample.shape
+    torch.Size([100, 2, 10])
+
+    >>> emp_dist.log_prob(single_sample).shape
+    torch.Size([2])
+    >>> # Vectorized samples cannot be scored by log_prob.
+    >>> with pyro.validation_enabled():
+    ...     emp_dist.log_prob(batch_sample).shape
+    Traceback (most recent call last):
+     ...
```

Comment:
I am not sure if 5 spaces here is right.

Reply:
Do you mean the `emp_dist` line? That has 4 spaces if you count w.r.t. the with block.

Reply:
I mean the line after Traceback. It has 5 spaces before "..."